### PR TITLE
Pass New File ID to Worker in Image Build

### PR DESCRIPTION
**Pull request by ThomasThelen**
Creates a file in the GRIDFS assetstore and passes the id to the worker. I didn’t realize one of the test files wasn’t running so I made some additional changes to get it working (See ` plugin.cmake & dataone_utils_test.py` below.

utils.py
   Added a method, create_repository_file, that creates a file on the GRIDFS assetstore and returns its id.
  Also modified getOrCreateRootFolder so that it can take a description.

  One of the unit tests caught an error-I should have been checking the mongo cursor for its `.count()` rather than checking if the cursor was bool. (line 62).

image.py
  Modified so that getOrCreateRootFolder is called and the file id is passed to the job.

plugin.cmake & dataone_utils_test.py
   I had previously moved these tests into the utils.py file, but never updated plugin.cmake. I removed the old test file and added the new entry for utils.py

### CODE REVIEW

File: server/utils.py [hunk 1]
```diff
@@ -194,3 +195,64 @@ def get_tale_files(tale, user):
...
+
+    """
+    Get all of the assetstores and retrieve GridFS. The reason we're using this
+     assetstore is that we don't want this folder in the user's home directory.
+    """
+    assetstores = ModelImporter.model('assetstore').list()
+    assetstore_id = int()
+
+    for store in assetstores:
+        if store['name'] == 'GridFS local':
```
Comment (Xarthisius) [inline server/utils.py]: I don't expect that name to change. However, could we grab a list of assetstores by type and just pick the first one that's GridFS?

Comment (ThomasThelen) [reply server/utils.py]: Good call-let me check it out

Comment (ThomasThelen) [reply server/utils.py]: Committed

File: server/utils.py [hunk 2]
```diff
@@ -194,3 +197,64 @@ def get_tale_files(tale, user):
...
+    if assetstore_id == int():
+        logger.warning('Failed to get assetstore')
+        return None
+
+    # Get the assetstore from its id
+    try:
+        store = ModelImporter.model('assetstore').load(assetstore_id)
+    except Exception as e:
+        logger.debug('Error loading assetstore: {}'.format(e))
+        return None
```
Comment (Xarthisius) [inline server/utils.py]: I think lines L223-L239 could be simplified a bit into:
```
store = ModelImporter.model('assetstore').findOne({'type': AssetstoreType.GRIDFS})
if store is None:
    logger.warning('No GridFS assetstore found')
    return None
```

**Comment by ThomasThelen** [comment]
Code for issue #108 is also in this PR

**Pull request merged**